pkg/networkservice/common/discover/server.go

```diff
@@ -115,14 +118,18 @@ func (d *discoverCandidatesServer) Close(ctx context.Context, conn *networkservi
 		// If it's an existing connection, the NSE name should be set. Otherwise, it's probably an API misuse.
 		return nil, errors.Errorf("network_service_endpoint_name is not set")
 	}
-	nse, err := d.discoverNetworkServiceEndpoint(ctx, nseName)
-	if err != nil {
-		return nil, err
-	}
-	u, err := url.Parse(nse.Url)
-	if err != nil {
-		return nil, errors.WithStack(err)
+
+	var u *url.URL
+
+	if nse, err := d.discoverNetworkServiceEndpoint(ctx, nseName); err == nil {
+		u, err = url.Parse(nse.Url)
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+	} else {
+		logger.Errorf("no endpoint found for Close: %v", conn)
 	}
+
 	return next.Server(ctx).Close(clienturlctx.WithClientURL(ctx, u), conn)
```

Comment:
Am I get it correctly that in case if endpoint is unreachable we will call next with expired ctx?

Reply:
We will call next with no client URL, so it will close NSMgr chain, Forwarder chain, NSMgr chain again and will get an error in connect.
It is not related somehow to the context expiration.

Reply:
Missed this, thank you.

Reply:
We have `ctx.Deadline() / 10` timeout for `discoverNetworkServiceEndpoint`, so it wouldn't be expired.